File: mobile/src/main/java/com/ourygo/ygomobile/util/PlatformUtil.java
package com.ourygo.ygomobile.util;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

import java.util.List;

/**
 * Create By feihua  On 2021/10/18
 */
public class PlatformUtil {
    public static final String PACKAGE_WECHAT = "com.tencent.mm";
    public static final String PACKAGE_MOBILE_QQ = "com.tencent.mobileqq";
    public static final String PACKAGE_QZONE = "com.qzone";
    public static final String PACKAGE_SINA = "com.sina.weibo";

    // 判断是否安装指定app
    public static boolean isInstallApp(Context context, String app_package){
        final PackageManager packageManager = context.getPackageManager();
        List<PackageInfo> pInfo = packageManager.getInstalledPackages(0);
        if (pInfo != null) {
            for (int i = 0; i < pInfo.size(); i++) {
                String pn = pInfo.get(i).packageName;
                if (app_package.equals(pn)) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: mobile/src/main/java/com/ourygo/ygomobile/util/PlatformUtil.kt
package com.ourygo.ygomobile.util

import android.content.Context

/**
 * Create By feihua  On 2021/10/18
 */
object PlatformUtil {
    const val PACKAGE_WECHAT = "com.tencent.mm"
    const val PACKAGE_MOBILE_QQ = "com.tencent.mobileqq"
    const val PACKAGE_QZONE = "com.qzone"
    const val PACKAGE_SINA = "com.sina.weibo"

    // 判断是否安装指定app
    fun isInstallApp(context: Context, app_package: String?): Boolean {
        val packageManager = context.packageManager
        val pInfo = packageManager.getInstalledPackages(0)
        if (pInfo != null) {
            for (i in pInfo.indices) {
                val pn = pInfo[i].packageName
                if (app_package == pn) {
                    return true
                }
            }
        }
        return false
    }
}